### PR TITLE
feat: auth and toast component

### PR DESCRIPTION
**Pull request by partha8**
## files to check
- [src/components/Toast/Toast.js](https://github.com/partha8/algern-clothing-react/compare/auth?expand=1#diff-0d3364c2392f3f257e14a5098264abe66ae412cfcbd99f44bee6a7a76032de7f)
- [src/context/AuthProvider.js](https://github.com/partha8/algern-clothing-react/compare/auth?expand=1#diff-ae160e02f5ac59fd6a714d684489d5b67435054d5ba2a3777ed10bd3ad587190)
- [src/pages/AuthPages/Login/Login.js](https://github.com/partha8/algern-clothing-react/compare/auth?expand=1#diff-631f0c32b16ed1371542fce45c16f442b8b56f7534c23102de722f6515fd856f)
- [src/pages/AuthPages/SignUp/SignUp.js](https://github.com/partha8/algern-clothing-react/compare/auth?expand=1#diff-271b5def633e27d3970b21c71e66d3db494c6e81eaaf2a19a6a79c248c9e2098)
- [src/reducers/authReducer.js](https://github.com/partha8/algern-clothing-react/compare/auth?expand=1#diff-978cb6ff885969efafb050f3963f42db8388343b0f752803094897fa970c7023)

## Note
The number of lines that need to be reviewed is less and it's not 689. I have just made some changes to the previous code, like making custom hooks to fetch products and categories which I took out from the state provider. These lines have been reviewed and approved, you can ignore them. **Please review only auth-related code**. Thank you.

### CODE REVIEW

File: src/components/Navbar/Navbar.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { GiHamburgerMenu } from "react-icons/gi";
 
 import { useStateContext } from "../../context/StateProvider";
 import { Link } from "react-router-dom";
+import { useAuthContext } from "../../context/AuthProvider";
```
Comment (nayyyhaa) [inline src/components/Navbar/Navbar.js:8]: just a suggestion, you can use absolute paths while importing files. Checkout this article: https://dev.to/emanuelone/how-to-use-absolute-path-in-reactjs-503i

File: src/pages/AuthPages/SignUp/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                name="confirmPassword"
+                onChange={handleChange}
+              />
+              <span>{error}</span>
+            </div>
+
+            {/* <div>
+              <label className="auth-label" for="show-password">
+                Show Password
+              </label>
```
Comment (nayyyhaa) [inline src/pages/AuthPages/SignUp/SignUp.js]: can remove commented code if not required anymore.

**Review by nayyyhaa — APPROVED** [review]
Appreciation: Code is well written & easy to understand. Proper usage of contexts & reducers are done. Nice work done! you can merge the code. 👍

File: src/components/Toast/Toast.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { useEffect } from "react";
+import { useStateContext } from "../../context/StateProvider";
+import "./toast.css";
+
+export const Toast = () => {
+  const { toast, toastHandler } = useStateContext();
+  const { showToast, message, type } = toast;
```
Comment (foolhardy21) [inline src/components/Toast/Toast.js:8]: just a suggestion - you can destructure `toast` in the same line as line 7.

File: src/components/Toast/Toast.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      toastHandler();
+    }, 3000);
+    return () => clearTimeout(timeout);
```
Comment (foolhardy21) [inline src/components/Toast/Toast.js:14]: good practice of clearing the `timeout`.

File: src/context/AuthProvider.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        );
+        toastHandler(
+          true,
+          `Welcome back ${response.data.foundUser.firstName}!`,
+          "success"
+        );
+        navigate("/profile");
+      }
+    } catch (error) {
+      console.error(error);
```
Comment (foolhardy21) [inline src/context/AuthProvider.js:49]: just a suggestion - you may later on add an alert for the error too.

File: src/pages/AuthPages/Login/Login.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+        <div className="login" id="login">
+          <h3>Welcome Back!</h3>
+
+          <form className="auth-form" onSubmit={loginHandler}>
+            <div>
+              <label className="auth-label">
+                Email Address<span className="req">*</span>
+              </label>
+              <input
```
Comment (foolhardy21) [inline src/pages/AuthPages/Login/Login.js:52]: suggestion - you may keep the input inside the label.

**Review by foolhardy21 — APPROVED** [review]
good work. good file structure. added a few suggestions.

File: src/components/Navbar/Navbar.js (hunk 1 above)
Comment (partha8) [reply src/components/Navbar/Navbar.js:8]: Yup, i did this in other imports, missed this one.

File: src/pages/AuthPages/SignUp/SignUp.js (hunk 2 above)
Comment (partha8) [reply src/pages/AuthPages/SignUp/SignUp.js]: Done.

**Pull request merged**